### PR TITLE
Added task to report different activities

### PR DESCRIPTION
**Pull request by tommygonzaleza**

### CODE REVIEW

**Comment by tommygonzaleza** [comment]
https://github.com/breatheco-de/breatheco-de/issues/6030

File: breathecode/assignments/serializers.py [hunk 1]
```diff
@@ -211,6 +212,24 @@ def validate(self, data):
                     'review status',
                     slug='editing-revision-status-but-is-not-teacher-or-assistant')
 
+        if 'opened_at' in data and data['opened_at'] > self.instance.opened_at:
```
Comment (jefer94) [inline breathecode/assignments/serializers.py:215]: if you wanna do this in the serializer, should be better placed it in the create or update method, this method just run the validation, this shouldn't be runned if it's runned in bulk mode, and one of them fail its checks, this endpoint I think does not have bulk mode support but this should break in a future

File: breathecode/assignments/views.py [hunk 2]
```diff
@@ -636,6 +637,10 @@ def post(self, request, user_id=None):
         if serializer.is_valid():
             serializer.save()
             # tasks.teacher_task_notification.delay(serializer.data['id'])
+            tasks_activity.add_activity.delay(request.user.id,
```
Comment (jefer94) [inline breathecode/assignments/views.py:640]: is it created when a student opens a module?

File: breathecode/authenticate/views.py [hunk 3]
```diff
@@ -401,6 +402,11 @@ def put(self, request, new_status=None):
                                                      last_name=item.last_name)
                     profile_academy.save()
 
+                tasks_activity.add_activity.delay(request.user.id,
+                                                  'invite_status_updated',
+                                                  related_type='auth.UserInvite',
+                                                  related_id=item.id)
+
```
Comment (jefer94) [inline breathecode/authenticate/views.py:409]: Use `serializer.is_valid` to make sure this activity will be registered when the data is ok

**Review by jefer94 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**